### PR TITLE
feat: local deployment

### PR DESCRIPTION
**Pull request by 0xkenj1**
# 🤖 Linear

Closes GIT-82

## Description
Set up for local deployment and short docs.

## Checklist before requesting a review

-   [x] I have conducted a self-review of my code.
-   [x] I have conducted a QA.
-   [x] If it is a core feature, I have included comprehensive tests.

### CODE REVIEW

File: apps/processing/Dockerfile
Comment (0xnigir1) [inline apps/processing/Dockerfile:1]: this file should be deleted right? is the Dockerfile for the Indexer not the processing service

File: apps/processing/.dockerignore
Comment (0xnigir1) [inline apps/processing/.dockerignore:1]: delete too

File: docker-compose.yaml [hunk 1]
```diff
@@ -56,20 +54,24 @@ services:
...
-            - db_data:/var/lib/envio-postgresql/data
+            - db_data:/var/lib/indexer-postgres/data
         env_file:
             - .env
         environment:
             POSTGRES_DB: ${ENVIO_PG_DATABASE}
             POSTGRES_USER: ${ENVIO_PG_USER}
             POSTGRES_PASSWORD: ${ENVIO_POSTGRES_PASSWORD}
+        healthcheck:
+            test: ["CMD-SHELL", "pg_isready -U ${DATALAYER_PG_USER}"]
```
Comment (0xnigir1) [inline docker-compose.yaml]:
```suggestion
            test: ["CMD-SHELL", "pg_isready -U ${ENVIO_PG_USER}"]
```

File: packages/indexer-client/src/providers/envioIndexerClient.ts [hunk 2]
```diff
@@ -69,7 +70,7 @@ export class EnvioIndexerClient implements IIndexerClient {
             if (error instanceof InvalidIndexerResponse) {
                 throw error;
             }
-            throw new IndexerClientError(JSON.stringify(error));
+            throw new IndexerClientError(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```
Comment (0xnigir1) [inline packages/indexer-client/src/providers/envioIndexerClient.ts:73]: 🥴 xd

File: docker-compose.yaml
Comment (0xnigir1) [inline docker-compose.yaml:1]: we should update .env.example file too

File: README.md [hunk 3]
```diff
@@ -19,15 +19,27 @@ Ensure you have the following installed on your machine:
...
 
-To deploy locally with Docker, run the following command:
+1. Build and start the services in detached mode:
 
 ```
 docker-compose up -d --build
 ```
 
-Once the deployment is complete, you can access Hasura by navigating to:
+2. After starting Docker Compose, run the following command to apply the database migrations:
```
Comment (0xnigir1) [inline README.md]: i think we should clarify that you need to setup the `.env` file inside `apps/scripts` folder before running the scripts

File: packages/indexer-client/src/providers/envioIndexerClient.ts (hunk 2 above)
Comment (0xkenj1) [reply packages/indexer-client/src/providers/envioIndexerClient.ts:73]: will be fixed on the following pr

**Review by jahabeebs — APPROVED** [review]

**Review by 0xnigir1 — APPROVED** [review]

**Pull request merged**